astropy/cosmology/_io/mapping.py

```diff
@@ -134,16 +134,27 @@
      'cosmo_name': 'Planck18', ...
 """
 
+from __future__ import annotations
+
 import copy
-from collections.abc import Mapping
+from collections.abc import Mapping, MutableMapping
+from typing import TYPE_CHECKING, Any, Literal, TypeVar
 
 from astropy.cosmology.connect import convert_registry
 from astropy.cosmology.core import _COSMOLOGY_CLASSES, Cosmology
 
-__all__ = []  # nothing is publicly scoped
+if TYPE_CHECKING:
+    from astropy.cosmology._typing import _CosmoT
+
+    _MapT = TypeVar("_MapT", MutableMapping[str, Any])
+
+
+__all__: list[str] = []  # nothing is publicly scoped
 
 
-def _rename_map(map, /, renames):
+def _rename_map(
+    map: Mapping[str, Any], /, renames: Mapping[str, str]
+) -> dict[str, Any]:
```

Comment:
Why couldn't we use something like
```suggestion
T = TypeVar("T")
...
def _rename_map(
    map: Mapping[str, T], /, renames: Mapping[str, str]
) -> dict[str, T]:
```

Reply:
`map` can be heterogeneous in the value

Reply:
If the `map` values are heterogeneous then `T` would just represent some union type. But having union types in the output isn't very helpful. Let's leave this as it is then.